src/index.js

```diff
@@ -476,6 +477,10 @@ class Node extends EventEmitter {
    * @param {PeerInfo} peerInfo
    */
   _peerDiscovered (peerInfo) {
+    if (peerInfo.id.toB58String() === this.peerInfo.id.toB58String()) {
```

Comment:
Is it more efficient to compare the underlying buffers? ie peerInfo.id.id.equals(this.peerInfo.id.id)

Reply:
It's actually not. The stringified version is already stored when the instance is created, so there's no conversion actually happening, it's a straight in-memory check. For the buffer check, if the buffers aren't strictly equivalent (the majority of cases) `.compare` is run to check each byte, which is slower.

Reply:
Got it 👍